src/Compilers/Test/Utilities/CSharp/CompilationTestUtils.cs

```diff
@@ -407,5 +408,40 @@ ExpressionSyntax asExpression(SyntaxNode node)
                 return expr;
             }
         }
+
+        private sealed class TopLevelNullabilityRetreiver : BoundTreeWalker
+        {
+            public Dictionary<SyntaxNode, TypeSymbolWithAnnotations> Map { get; set; }
```

Comment:
>TypeSymbolWithAnnotations [](start = 42, length = 25)

I expected we would retrieve states, or states+annotations (to capture r-value and l-value), not annotations alone

Reply:
I'm just adapting this to the existing pattern, which uses TWSA to verify everything. When the actual public api is added, I expect to be ripping this out entirely and doing verification based solely on the public API.

---
In reply to: [264941171](https://github.com/dotnet/roslyn/pull/34017#discussion_r264941171) [](ancestors = 264941171)